plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaPageSink.java

```diff
@@ -126,7 +126,13 @@ public CompletableFuture<?> appendPage(Page page)
             keyBytes = keyEncoder.toByteArray();
             messageBytes = messageEncoder.toByteArray();
 
-            expectedWrittenBytes += keyBytes.length + messageBytes.length;
+            /**
+             * KafkaProducerBatch returns -1 for bytes length if data is null
+             * @see org.apache.kafka.clients.producer.internals.ProducerBatch#tryAppend
+             */
```

Comment:
Was this causing some user visible defect? Can you have a test for that? Is it possible that `finish()` method would fail because of incorrect value in `expectedWrittenBytes`, how can you trigger that to test your fix?

Reply:
Will update: this did cause test failures w empty field handling, will call it out explicitly, thanks!

Reply:
The init methods fail for BaseConnectorTest and BaseConnectorSmokeTest. Here is the error without this change:
```
[ERROR] Tests run: 724, Failures: 1, Errors: 0, Skipped: 425, Time elapsed: 122.4 s <<< FAILURE! -- in TestSuite
[ERROR] io.trino.plugin.kafka.TestKafkaWithConfluentLatestConnectorSmokeTest.init -- Time elapsed: 13.61 s <<< FAILURE!
io.trino.testing.QueryFailedException: Cannot read the array length because "keyBytes" is null
```